### PR TITLE
Add LinearResistance equation test

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes part of https://github.com/Deltares/Ribasim/issues/206. I'm giving the feature branch a try.

### CODE REVIEW

File: core/test/equations.jl [hunk 1]
```diff
@@ -44,3 +44,19 @@ TimerOutputs.disable_debug_timings(Ribasim)  # causes recompilation (!)
...
+    @test ispath(toml_path)
+    model = Ribasim.run(toml_path)
+    p = model.integrator.p
+
+    t = Ribasim.timesteps(model)
+    storage = Ribasim.get_storages_and_levels(model).storage[1, :]
+    basin_area = p.basin.area[1][2] # Considered constant
+    cons1 = 450.0 # Limit storage 
+    cons2 = -1 / (basin_area * p.linear_resistance.resistance[1]) # decay rate
+    storage_analytic = cons1 .+ (storage[1] - cons1) .* exp.(cons2 .* t)
```
Comment (Hofer-Julian) [inline core/test/equations.jl:59]:
```suggestion
    storage_analytic = cons1 + (storage[1] - cons1) * exp(cons2 * t)
```
All the values are scalars, right?
So I guess you don't need the dots

File: core/test/equations.jl [hunk 2]
```diff
@@ -44,3 +44,19 @@ TimerOutputs.disable_debug_timings(Ribasim)  # causes recompilation (!)
...
+    toml_path = normpath(@__DIR__, "../../data/linear_resistance/linear_resistance.toml")
+    @test ispath(toml_path)
+    model = Ribasim.run(toml_path)
+    p = model.integrator.p
+
+    t = Ribasim.timesteps(model)
+    storage = Ribasim.get_storages_and_levels(model).storage[1, :]
+    basin_area = p.basin.area[1][2] # Considered constant
+    cons1 = 450.0 # Limit storage 
+    cons2 = -1 / (basin_area * p.linear_resistance.resistance[1]) # decay rate
```
Comment (Hofer-Julian) [inline core/test/equations.jl:58]: Maybe call the variable `decay_rate` then?

File: core/test/equations.jl [hunk 3]
```diff
@@ -44,3 +44,19 @@ TimerOutputs.disable_debug_timings(Ribasim)  # causes recompilation (!)
...
+@testset "LinearResistance" begin
+    toml_path = normpath(@__DIR__, "../../data/linear_resistance/linear_resistance.toml")
+    @test ispath(toml_path)
+    model = Ribasim.run(toml_path)
+    p = model.integrator.p
+
+    t = Ribasim.timesteps(model)
+    storage = Ribasim.get_storages_and_levels(model).storage[1, :]
+    basin_area = p.basin.area[1][2] # Considered constant
+    cons1 = 450.0 # Limit storage
```
Comment (Hofer-Julian) [inline core/test/equations.jl:57]: Where does this number come from?

**Review by Hofer-Julian — COMMENTED** [review]
Looks mostly good to me

File: core/test/equations.jl (hunk 1 above)
Comment (SouthEndMusic) [reply core/test/equations.jl:59]: No, this is the storage over time

File: core/test/equations.jl (hunk 3 above)
Comment (SouthEndMusic) [reply core/test/equations.jl:57]: See added comment

File: core/test/equations.jl (hunk 1 above)
Comment (Hofer-Julian) [reply core/test/equations.jl]: I see, `t` is a vector

**Review by Hofer-Julian — APPROVED** [review]

**Comment by visr** [comment]
> I'm giving the feature branch a try.

How should this be used, is this part of a series of PRs you want to do on the linear_resistance branch? This looks good to merge onto main, but how do we get it there?

**Comment by SouthEndMusic** [comment]
> How should this be used, is this part of a series of PRs you want to do on the linear_resistance branch? This looks good to merge onto main, but how do we get it there?

The idea was to make a PR to the `[Test-node-equations-feature](https://github.com/Deltares/Ribasim/tree/Test-node-equations-feature)` branch for every node type / equation, and then merge to main once all equations are done. I see that that is not necessary in this case (it would have been nice for the PID control branch I think).

**Comment by Hofer-Julian** [comment]
>  This looks good to merge onto main, but how do we get it there?

@SouthEndMusic For the future, the target branch of a PR can still be changed after the PR has been opened :)

**Pull request merged**